Automattic/newspack-plugin
# Registration block – login with Google

### All Submissions:






### Changes proposed in this Pull Request:



Adds Google Login to the Reader Registration block. Also adds a `np_reader_login_method` user meta, to note the login method. 

<img width="541" alt="image" src="https://user-images.githubusercontent.com/7383192/178972096-dd005e1c-c63f-498d-a30b-ca23c5c003d7.png">

This implementation reuses a lot of the code used for processing admin-related Google OAuth used for GAM connection. The differences are:

- only `auth/userinfo.email` scope is requested 
- credentials are not saved, since there will be no actions performed on behalf of the user

Closes #1774 

### How to test the changes in this Pull Request:

1. Without configuring Google OAuth proxy (no `NEWSPACK_GOOGLE_OAUTH_PROXY` defined), observe that the Registration block does _not_ feature a "Sign in with Google" button
2. Configure Google OAuth proxy by providing a `NEWSPACK_GOOGLE_OAUTH_PROXY` env. variable
3. Observe the Reader Registration block allows for signing in via Google
4. After being authenticated, observe the "My Account" page is accessible
5. Observe in the DB that the `np_reader_login_method` user meta value is `'google'`

### Other information:
